### ISSUE TITLE
versionCompare: Piping into echo

### ISSUE THREAD

**Issue by rwin-novo** [yousefvand/shellman#29]
Hi,

I'm really glad I've found your project, it's really nice!

I'm really trying to understand some parts of it, but that's my hurdle, as the code works (mostly, see point 3). My IDE (IntelliJ) is complaining though for some parts. Can you please explain this block?

```
function versionCompare () {
  function subVersion () {
    local len=${#1}
    temp=${1%%"."*} && indexOf=$(echo ${1%%"."*} | echo ${#temp})
    echo -e "0:indexOf"
  }
```

My questions:

1. len is unused. is it needed?
2. what is the purpose of piping into an echo (which is not supported afaik)?
3. the `echo -e "0:indexOf"` part seems to be wrong. in an older version, you had `echo -e "${1:0:indexOf}"`, which worked, but this gives `bash: ((: 0:indexOf: syntax error in expression (error token is ":indexOf")`. A bug?

Thanks
rw

**Comment by yousefvand** [comment]
Hi
- No it is not needed. Probably left there from older versions (bug). Thanks for heads-up.
If you are willing you can send a PR. [here is the code](https://github.com/yousefvand/shellman/blob/master/nsroot/fn-fx/fn-version-compare.json#L10)
- Honestly I don't know. Probably CTRL+C/CTRL+V from somewhere. But without it I get error.
In Shellman I just coupled big parts and organized them, small parts like this one come from others.
If you came up with a cleaner version of piping to echo, I'd happily merge your PR.
- Which version you are using? In latest one it is [`echo -e "${1:0:indexOf}`](#https://github.com/yousefvand/shellman/blob/master/nsroot/fn-fx/fn-version-compare.json#L12) and works fine.
Since Shellman primarily is written for vscode and you mentioned IntelliJ, I guess you need to inform the maintainer of Shellman for IntelliJ. I'm not aware if both IDEs support the same format for snippests or a conversion is needed.

**Comment by rwin-novo** [comment]
Thanks! 

1. I'll send a PR (pinky promise).
2. Ehm, as admitted, I'm sometimes having hard time to interpret stuff so don't have any better version atm.
3. I copied it from https://github.com/yousefvand/shellman/blob/master/COMMANDS.md#fn-version-compare--fn-semver-compare . Other lines seem to be affected as well. But it's probably just the markdown renderer and not the code itself?!

**Comment by yousefvand** [comment]
> 3. I copied it from https://github.com/yousefvand/shellman/blob/master/COMMANDS.md#fn-version-compare--fn-semver-compare . Other lines seem to be affected as well. But it's probably just the markdown renderer and not the code itself?!

Yes just `COMMANDS.md` is affected and I was aware of that. The snippet produces right code.
I had a rough time (multiple commits to fix this) to render snippet's `JSON` to shellscript code for `COMMANDS.md` but to no avail.
My rendering code is in `build.js`, the part mentions "generating documentation" (formerly docGen function).
`build.go` doesn't implement document generation yet.

**Comment by yousefvand** [comment]
Closed by v5.5.0
